modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -97,14 +318,23 @@ ModularGapConductanceConstraint::computeQpResidual(Moose::MortarType mortar_type
         }
 
       // compute an ADReal gap width to pass to each gap flux model
-      const auto gap_width = (ad_phys_points_primary - ad_phys_points_secondary) * _normals[_qp];
+      _gap_width = (ad_phys_points_primary - ad_phys_points_secondary) * _normals[_qp];
+
+      // First, compute radii with _gap_width
+      computeGapRadii(_gap_width);
+
+      // With radii, compute adjusted gap length for conduction
+      _adjusted_length = computeGapLength();
```

Comment:
same here, the function could just set them member

Reply:
Same idea, we see what we are doing here without diving into the member function. Then the member function can be `const`, making sure we are not messing other stuff.

Reply:
ok